src/pages/api/nostr/post-event.ts

```diff
@@ -195,120 +195,65 @@ const PostEvent = async (req: NextApiRequest, res: NextApiResponse) => {
   if (req.method !== 'POST') {
     return res.status(405).json({});
   }
-
   try {
     const event = parseRequestBody(req.body);
     const privkey = event.privkey;
     delete event.privkey;
 
     const kind = event.kind;
-
     const relays = event.relays;
     delete event.relays;
-
     const pool = new SimplePool();
+    
+    let signedEvent;
 
     if (kind === 1) {
-      const signedEvent = finishEvent(event, privkey);
-  
-      let sub = pool.sub(relays, [
-        {
-          kinds: [kind],
-          authors: [event.pubkey],
-        },
-      ]);
-  
-      sub.on('event', (event) => {
-        console.log('got event:', event);
-      });
-  
-      await pool.publish(relays, signedEvent);
-  
-      let events = await pool.list(relays, [{ kinds: [0, kind] }]);
-      let postedEvent = await pool.get(relays, {
-        ids: [event.id],
-      });
+      signedEvent = finishEvent(event, privkey);
+
     } else if (kind === 4) {
       let sk1 = privkey;
       let pk1 = event.pubkey;
-      
       let pk2 = event.tags[0][1];
-
       let ciphertext = await nip04.encrypt(sk1, pk2, event.content);
-  
       let nip04Event = {
         kind: kind,
         pubkey: pk1,
         tags: [['p', pk2]],
         content: ciphertext,
         created_at: Math.floor(Date.now() / 1000),
       };
-
-      const signedEvent = finishEvent(nip04Event, privkey);
-
-      let sub = pool.sub(relays, [
-        {
-          kinds: [kind],
-          authors: [event.pubkey],
-        },
-      ]);
-  
-      sub.on('event', (event) => {
-        console.log('got event:', event);
-      });
-
-      await pool.publish(relays, signedEvent);
       
-      let events = await pool.list(relays, [{ kinds: [0, kind] }]);
-      let postedEvent = await pool.get(relays, {
-        ids: [event.id],
-      });
+      signedEvent = finishEvent(nip04Event, privkey);
```

Comment:
I see finishEvent in all of these event types. Can we extract that out to somewhere? Also do we need all of these types? @calvadev

Reply:
We need finishEvent to be called independently since each event corresponding to a different kind is constructed differently. As for the other deleted code, really only `await pool.publish(relays, signedEvent);` needs to be added back. The sub calls just monitor the relay list to see when the event is posted to the relays and basically the same goes for pool.list and pool.get.

Reply:
I fixed the ordering of the postevents function and pulled the finish events out. LMK if the logic makes sense for you. I think this makes the function alot cleaner and still the same functionality. Also the `await pool.publish(relays, signedEvent);` is after the if statements